scripts-dev/config-lint.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Find linting errors in Synapse's default config file.
+# Exits with 0 if there are no problems, or another code otherwise.
+
+# Fix non-lowercase true/false values
+sed -i -E "s/(#.*): +True(.*)/\1: true\2/g; s/(#.*): +False(.*)/\1: false\2/g;" synapse/config/*.py
```

Comment:
1. why do we require a #
2. some of the groups are redundant:
```suggestion
sed -i -E "s/(#.*): +True/\1: true/g; s/(#.*): +False/\1: false/g;" synapse/config/*.py
```

Reply:
The code can contain false positives without the hash, such as docstrings:

https://github.com/matrix-org/synapse/blob/b617864cd9f81109e818bc5ae95bee317d917b72/synapse/config/logger.py#L199

Reply:
ahhh. we should probably run it on the sample config rather than attempt to match only the config bits within `synapse/config`.